src/main.rs

```diff
@@ -233,14 +250,25 @@ fn build_sysroot(
 
     // Determine where to put the sysroot.
     let user_dirs = directories::ProjectDirs::from("de", "ralfj", "cargo-careful").unwrap();
-    let sysroot_dir = user_dirs.cache_dir();
+    let mut sysroot_dir: PathBuf = user_dirs.cache_dir().to_owned();
+
+    if let Some(san) = sanitizer {
+        sysroot_dir.push(san)
+    }
 
     // Do the build.
-    eprint!("Preparing a careful sysroot (target: {target})... ");
+    eprint!(
+        "Preparing a careful sysroot (target: {target}{})... ",
+        if let Some(san) = sanitizer {
+            format!(", sanitizer: {san}")
+        } else {
+            String::new()
+        }
```

Comment:
```suggestion
        if let Some(san) = sanitizer {
            format_args!(", sanitizer: {san}")
        } else {
            format_args!("")
        }
```
this should work just as well and avoids intermediate allocations